Source/DataSources/CzmlDataSource.js

```diff
@@ -1794,7 +1801,17 @@ define([
         processPacketData(Number, polyline, 'granularity', polylineData.granularity, interval, sourceUri, entityCollection);
         processMaterialPacketData(polyline, 'material', polylineData.material, interval, sourceUri, entityCollection);
         processMaterialPacketData(polyline, 'depthFailMaterial', polylineData.depthFailMaterial, interval, sourceUri, entityCollection);
-        processPacketData(Boolean, polyline, 'followSurface', polylineData.followSurface, interval, sourceUri, entityCollection);
+        // Don't want to break CZML spec to keep this workaround even after followSurface has been deprecated from geometry.
+        // See https://github.com/AnalyticalGraphicsInc/cesium/pull/7582#discussion_r258695385
+        if (defined(polylineData.followSurface) && !defined(polylineData.arcType)) {
+            if (polyline.followSurface) {
+                processPacketData(ArcType, polyline, 'arcType', ArcType.GEODESIC, interval, sourceUri, entityCollection);
```

Comment:
In addition to several coding errors, this approach won't work in general, since `followSurface` could be time-varying.  Rather than turning a `boolean` into an `ArcType`, this needs to turn a `Property<boolean>` into a `Property<ArcType>` which is essentially `map` since `Property` objects are basically functions.